src/NewTools-Debugger-Tests/StDebuggerExtensionMechanismTest.class.st

```diff
@@ -77,11 +131,22 @@ StDebuggerExtensionMechanismTest >> testRegisterExtensionTool [
 
 { #category : #'tests - extensions' }
 StDebuggerExtensionMechanismTest >> testUsesExtensions [
+
 	| testClass |
-	testClass := StDebuggerExtensionGeneratorForTests testClass.	
-	testClass showInDebugger: true.
-	self debuggerClass activateDebuggerExtensions: false.
+	testClass := StDebuggerExtensionGeneratorForTests testClass.
+	
+	self debuggerClass extensionToolsClasses do: [ :extension |
+		extension showInDebugger: false ].
+
+	self assertEmpty: self debuggerClass activatedDebuggerExtensions.
 	self deny: self debuggerClass usesExtensions.
-	self debuggerClass activateDebuggerExtensions: true.
-	self assert: self debuggerClass usesExtensions
+	self debuggerClass extensionToolsClasses ifNotEmpty: [
+		| activatedExtension |
+		activatedExtension := self debuggerClass extensionToolsClasses
+			                      asOrderedCollection first.
+		activatedExtension showInDebugger: true.
+		self
+			assertCollection: self debuggerClass activatedDebuggerExtensions
+			hasSameElements: { activatedExtension }.
+		self assert: self debuggerClass usesExtensions ]
```

Comment:
I don't know. I don't think it's possible and that's why we do this.

Because we can't ensure that there is a debugger extension in the image.

And we can't use any debugger extension for test because `StDebugger class>>#activatedDebuggerExtensions` rejects them.

Initially the test was about activating/unactivating the `activateDebuggerExtensions` setting, which doesn't exist anymore. Maybe the test should just be removed.

Or, we need to generate an extension class on the fly. But it needs to inherit from `SpPresenter`, to define a layout and the explicit requirements from `TStDebuggerExtension`. I haven't managed to do it and that's why I created the debugger extension for tests, which was previously generated.
It is possible that I don't know how to compile methods correctly though